website/notifications/utils.py

```diff
@@ -426,14 +429,17 @@ def subscribe_user_to_notifications(node, user):
             global_subscription = NotificationSubscription.load(global_event_id)
 
             subscription = NotificationSubscription.load(event_id)
-            if not subscription:
-                subscription = NotificationSubscription(_id=event_id, owner=node, event_name=event)
-            if global_subscription:
-                global_notification_type = get_global_notification_type(global_subscription, user)
-                subscription.add_user_to_subscription(user, global_notification_type)
-            else:
-                subscription.add_user_to_subscription(user, notification_type)
-            subscription.save()
+
+            # if no subscription for component and creator is the user, do not create subscription
+            if not(node and node.parent_node and not subscription and node.creator == user):
```

Comment:
Also minor: it might be worth mentioning in a code comment that "no subscription" = "adopt from parent". As you know, this isn't immediately obvious.